File: 더맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    q = scoville
    now = 0
    while q:
        mini = heapq.heappop(q)
        if mini>=K:
            break
        if q: next_mini = heapq.heappop(q)
        else: break
        now = mini + 2*next_mini
        answer +=1
        heapq.heappush(q, now)
    if now<K:
        answer = -1
    return answer

File: 더맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> q= new PriorityQueue<>();
        Arrays.stream(scoville).forEach(s-> q.add(s));
        int first = q.poll();
        while(first<K && q.size()>0){
            int mix = first+(q.poll()*2);
            q.add(mix);
            answer++;
            first = q.poll();               
        }
        if (first<K) answer = -1;
        
        return answer;
    }
}
